drafts/str4d-stratum/draft1.rst

```diff
@@ -0,0 +1,256 @@
+
+When displaying the current target in the UI to users, miners MAY convert the
+target to an integer difficulty as used in Bitcoin miners. When doing so, miners
+SHOULD use ``powLimit`` (as defined in ``src/chainparams.cpp``) as the basis for
+conversion.
+
+``mining.notify()``
+-------------------
+
+Server message::
+
+    {"id": null, "method": "mining.notify", "params": ["JOB_ID", "VERSION", "PREVHASH", "MERKLEROOT", "RESERVED", "TIME", "BITS", CLEAN_JOBS]}\n
+
+- ``JOB_ID`` (str): The id of this job.
+
+- ``VERSION`` (int): The block header version. Used as a switch for subsequent
+  parameters.
+
+The following parameters are only valid for ``VERSION == 4``:
+
+- ``PREVHASH`` (hex): The hash of the previous block.
+
+- ``MERKLEROOT`` (hex): The Merkle root of the transactions in this block.
+
+- ``RESERVED`` (hex): A 256-bit reserved field; zero by convention.
+
+- ``TIME`` (hex): The block time suggested by the server.
+
+- ``BITS`` (compactBits): The current network difficulty target.
+
+- ``CLEAN_JOBS`` (bool): If true, a new block has arrived. The miner SHOULD
+  abandon all previous jobs.
+
+``mining.submit()``
+-------------------
+
+Request::
+
+    {"id": 4, "method": "mining.submit", "params": ["WORKER_NAME", "JOB_ID", "TIME", "NONCE_2", "EQUIHASH_SOLUTION"]}\n
+
+- ``WORKER_NAME`` (str): A previously-authenticated worker name.
+
+- ``JOB_ID`` (str): The id of the job this submission is for.
+
+  - Miners MAY make multiple submissions for a single job id.
+
+- ``TIME`` (hex): The block time used in the submission.
+
+  - MAY be enforced by the server to be unchanged.
+
+- ``NONCE_2`` (hex): The second part of the block header nonce.
+
+Result::
+
+    {"id": 4, "result": ACCEPTED, "error": null}\n
+
+- ``ACCEPTED`` (bool): Whether the block was accepted.
+
+The server MAY provide an error message describing the reason for not accepting
+the block, if any.
+
+``client.reconnect()``
+----------------------
+
+Server message::
+
+    {"id": null, "method": "client.reconnect", "params": [("HOST", PORT, WAIT_TIME)]}\n
+
+- ``HOST`` (str): The host to reconnect to.
+
+  - ``pool.example.com``
+
+- ``PORT`` (int): The port to reconnect to.
+
+  - ``3337``
+
+- ``WAIT_TIME`` (int): Time in seconds that the miner should wait before
+  reconnecting.
+
+If ``client.reconnect`` is sent with empty parameters, the miner SHOULD
+reconnect to the same host and port it is currently connected to.
+
+``mining.suggest_target()``
+---------------------------
+
+Request (optional)::
+
+    {"id": 3, "method": "mining.suggest_target", "params": ["TARGET"]}\n
+
+- ``TARGET`` (hex): The target suggested by the miner for the next received job
+  and all subsequent jobs (until the next time this message is sent).
+
+The server SHOULD reply with ``mining.set_target``. The server MAY set the
+result id equal to the request id.
+
+
+Motivation
+==========
+
+Many existing cryptocurrency miners and pools use the original Stratum protocol
```

Comment:
And see http://docutils.sourceforge.net/docs/user/rst/quickref.html#citations for the syntax of citations in rST. (Put the citation definition wherever is convenient, it will be moved to the end of the document.)